frontend/app/views/spree/checkout/_delivery.html.erb

```diff
@@ -38,6 +38,7 @@
           <h5 class="stock-shipping-method-title"><%= Spree.t(:shipping_method) %></h5>
           <ul class="field radios shipping-methods">
             <% ship_form.object.shipping_rates.each do |rate| %>
+              <% next unless ship_form.object.selected_shipping_rate == rate || rate.shipping_method.available_to_users? %>
```

Comment:
I think this logic could be too complex for a view. What about creating a method into `Spree::ShippingRate` similar to:

```ruby
def available_to_user?
  shipping_method.available_to_users? || shipment.selected_shipping_rate == self
end
```

and here just do:

```erb
<% ship_form.object.shipping_rates.select(&:available_to_user?).each do |rate| %>
```

Do you think something like this would work and reduce view complexity?